python/semantic_kernel/connectors/ai/open_ai/prompt_execution_settings/azure_chat_prompt_execution_settings.py

```diff
@@ -12,68 +14,81 @@
+        use_enum_values=True,
+        extra="allow",
+    )
 
 
-@dataclass
-class ApiKeyAuthentication:
-    type: Literal["APIKey"] = "APIKey"
+class ConnectionStringAuthentication(AzureChatRequestBase):
+    type: Annotated[Literal["ConnectionString", "connection_string"], AfterValidator(to_snake)] = "connection_string"
+    connection_string: Optional[str] = None
+
+
+class ApiKeyAuthentication(AzureChatRequestBase):
+    type: Annotated[Literal["APIKey", "api_key"], AfterValidator(to_snake)] = "api_key"
     key: Optional[str] = None
 
 
-@dataclass
-class AzureEmbeddingDependency:
-    type: Literal["DeploymentName"] = "DeploymentName"
-    deploymentName: Optional[str] = None
+class AzureEmbeddingDependency(AzureChatRequestBase):
+    type: Annotated[Literal["DeploymentName", "deployment_name"], AfterValidator(to_snake)] = "deployment_name"
+    deployment_name: Optional[str] = None
+
+
+class DataSourceFieldsMapping(AzureChatRequestBase):
+    title_field: Optional[str] = None
+    url_field: Optional[str] = None
+    filepath_field: Optional[str] = None
+    content_fields: Optional[List[str]] = None
+    vector_fields: Optional[List[str]] = None
+    content_fields_separator: Optional[str] = "\n"
 
 
-@dataclass
-class AzureDataSourceParameters:
-    indexName: str
-    indexLanguage: Optional[str] = None
-    fieldsMapping: Dict[str, Any] = Field(default_factory=dict)
-    inScope: Optional[bool] = True
-    topNDocuments: Optional[int] = 5
-    semanticConfiguration: Optional[str] = None
-    roleInformation: Optional[str] = None
+class AzureDataSourceParameters(AzureChatRequestBase):
+    index_name: str
+    index_language: Optional[str] = None
+    fields_mapping: Optional[DataSourceFieldsMapping] = None
+    in_scope: Optional[bool] = True
+    top_n_documents: Optional[int] = 5
+    semantic_configuration: Optional[str] = None
+    role_information: Optional[str] = None
     filter: Optional[str] = None
-    embeddingKey: Optional[str] = None
-    embeddingEndpoint: Optional[str] = None
-    embeddingDeploymentName: Optional[str] = None
     strictness: int = 3
-    embeddingDependency: Optional[AzureEmbeddingDependency] = None
+    embedding_dependency: Optional[AzureEmbeddingDependency] = None
 
 
-@dataclass
-class AzureCosmosDBDataSource(AzureDataSourceParameters):
+class AzureCosmosDBDataSourceParameters(AzureDataSourceParameters):
     authentication: Optional[ConnectionStringAuthentication] = None
-    databaseName: Optional[str] = None
-    containerName: Optional[str] = None
-    embeddingDependencyType: Optional[AzureEmbeddingDependency] = None
+    database_name: Optional[str] = None
+    container_name: Optional[str] = None
+    embedding_dependency_type: Optional[AzureEmbeddingDependency] = None
 
 
-@dataclass
-class AzureAISearchDataSources(AzureDataSourceParameters):
+class AzureCosmosDBDataSource(AzureChatRequestBase):
+    type: Literal["azure_cosmos_db"]
```

Comment:
should we add a default here? `type: Literal['azure_cosmos_db'] = 'azure_cosmos_db'`?

Reply:
Yes, I think we probably should add this.  I can add it in one of the next PRs I'm going to work on, either for adding documentation for samples or for the pydantic settings.

Reply:
Actually, I went ahead with the change, since I had to merge from main anyway.  :)